integration-tests/docker/druid.sh

```diff
@@ -82,14 +83,18 @@ setupData()
   # The "query" and "security" test groups require data to be setup before running the tests.
   # In particular, they requires segments to be download from a pre-existing s3 bucket.
   # This is done by using the loadSpec put into metadatastore and s3 credientials set below.
-  if [ "$DRUID_INTEGRATION_TEST_GROUP" = "query" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "security" ]; then
+  if [ "$DRUID_INTEGRATION_TEST_GROUP" = "query" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "query-retry" ] || [ "$DRUID_INTEGRATION_TEST_GROUP" = "security" ]; then
```

Comment:
Don't have to be in this PR but maybe we should make this a flag passed in my mvn (like -Dstart.hadoop.docker=true). So something like -Ddruid.preload.test.data=true
So that this list doesn't get long in the future.

Reply:
Hmm, do you think there will be more test groups like these in the future? I'm not sure about it. A downside of using a flag is that you have to add the flag when you run a test manually. Also you have to be aware of that flag when you add a new test. If your concern is about code complexity, then we could provide a developer-friendly configuration where we can easily add or remove test groups which our docker script can handle in a more structured way.

Reply:
I am imaging the flags as easy control to the Druid cluster. Basically, there's a list of all the flags on the README and if you are writing a new test group, you can easily pick and choose how you want the state of your Druid cluster to be. i.e. you need to start x service then set this flag, you need pre-load data set this flag, you need hadoop set this flag, you want indexer instead of MM set this flag, etc.

Reply:
What is the main use case of the flag system? As what each test group needs are fixed (e.g., hadoop cluster must be started for hadoop task test), I'm not sure how it's better compared to that the docker script automatically launches necessary things depending on the test group.

Reply:
I believe it might be easier for a person who does not know the internal working of our Docker/Compose scripts to write new integration tests. Basically you do not need to know where all the bit and pieces are like to preload the data you need to modify the if check in druid.sh, and to do other stuff maybe we in other places, etc. 

For example, if I want to write a new integration test group that requires pre load data and hadoop running and maybe indexer running instead of MM, all of that might be in 3 different places in different files. With the flag system, you just need to specify what you want in the travis file for the mvn cmd.

Reply:
> If your concern is about code complexity, then we could provide a developer-friendly configuration where we can easily add or remove test groups which our docker script can handle in a more structured way.

Yeah, this comment above was to provide developer-friendly configurations so that they can easily add new integration tests. I guess what I think is pretty similar to yours, something like a centralized configuration system where you can define what you need for your integration test so that you don't have to worry about flags or configs at test run time. What I'm not sure is if they should be runtime flags. Maybe they could be if it's easier to implement.